### PR TITLE
mgr/dashboard: fix query parameters in task annotated endpoints

### PR DESCRIPTION
**Pull request by rjfd**
Fixes: http://tracker.ceph.com/issues/25096

I've also added a unit test to capture the issue reported above.

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/controllers/__init__.py [hunk 1]
```diff
@@ -268,24 +294,23 @@ def __init__(self, name, metadata, wait_for=5.0, exception_handler=None):
...
-            sig = getargspec(func)
-            arg_list = [a for a in sig.args]
+        params = _get_function_params(func)
 
-        for idx, arg in enumerate(arg_list):
+        args = args[1:]  # exclude self
+        for idx, param in enumerate(params):
             if idx < len(args):
-                arg_map[arg] = args[idx]
+                arg_map[param['name']] = args[idx]
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/controllers/__init__.py]: Too many ]s and [s. Can we create a new named tuple `FuncParam` in `_get_function_params`?
```python

from collections import namedtuple
FuncParam = namedtuple('FuncParam', 'name default required')
func_params.append(FuncParam(name=param , default=val, required=False)
```

Comment (rjfd) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: I could but that would require a lot of changes which are out of scope of this fix. I suggest you to create an issue for that.

Comment (rjfd) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: The `_get_function_params` function is used in more 6 places throughout the code.

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: Exactly. That's why we need to return proper types here.

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: Right now, this issue is just an inconvenience to @tspmelo . Merging this as it is would introduce further technical dept.

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: (Btw, I'm not blocking this PR in github)

**Comment by sebastian-philipp** [comment]
The fix itself works as expected.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-07-28_01:50:42-rados-wip-kefu-testing-2018-07-27-2326-distro-basic-mira
- http://pulpito.ceph.com/kchai-2018-07-30_02:08:39-rados-wip-kefu-testing-2018-07-27-2326-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-07-30_05:48:06-rados-wip-kefu-testing-2018-07-29-1514-distro-basic-smithi/

**Pull request merged**